### ISSUE TITLE
dummy DebugListener

### ISSUE THREAD

**Issue by AdrienPensart** [supermihi/pytaglib#23]
Hello, again me,

I have a big collection so when I'm scanning it, I see a lot of warning output generated by 

taglib/toolkit/tdebuglistener.cpp

do you think you can implement a dummy debug listener that do not output those warnings and call the setDebugListener with Python (I never did any pyx programming) ?

**Comment by supermihi** [comment]
Sounds like a good idea. I think it's not that hardcore, because we do not need to expose the dummy debug listener to python - it can purely live in the C++ world. Ideally, of course, it would somehow redirect the debug output into a python stream ... I will have a look what is possible without too much effort.

**Comment by supermihi** [comment]
Okay I played around a little but did not find any file that would issue a debug message to the console. After looking at the sources, it seems that debug output is enabled only if you compile taglib in DEBUG mode; otherwise, cmake sets the `NDEBUG` preprocessor flag which will silence the default debug listener.

Since most users use release builds of taglib (and if they don't, they probably know why) I think it is actually better to not silence the debug messages via pytaglib. Do you agree? Can you confirm that you are using a debug build?

**Comment by AdrienPensart** [comment]
Yep but actually, I have an "consistency" command in my program, supposed to show every warning in the music collection, so I want to choose when those messages are needed or not...so I think I'll need to fork taglib...

**Comment by AdrienPensart** [comment]
TagLib declared virtual methods for DebugListener, etc, but in the end, their debug function is commented out by NDEBUG, so stupid :)

**Comment by supermihi** [comment]
I've just pushed my progress to a new branch; see
https://github.com/supermihi/pytaglib/tree/silenceDebug
It compiles & runs but I didn't test if it actually silences the debug output. But at last you can see there that it's not too complicated to add custom C++ classes to Cython. :)

**Comment by AdrienPensart** [comment]
I did exactly the same code, but I missed that : 

```
+cdef void init():
 +    cdef ctypes.PytaglibDebugListener *_dl = new ctypes.PytaglibDebugListener()
 +    ctypes.setDebugListener(_dl)
 +
 +init()
```

Good job !

**Comment by supermihi** [comment]
Fixed in  f85896d.
